building/trusted_applications.rst

```diff
@@ -338,6 +338,69 @@ sequence:
         }
     }
 
+
+Identifying TA's client
+***********************
+
+The GP TEE specification is designed to ensure that TEE sessions are reliable
+once created. A TA instance can identify its client login method when a session
+is opened. A TA can use the client login credentials to establish or reject
+the session. A TA can get its client identity from property
+``"gpd.client.identity"`` with the TEE Internal Core API function
+``TEE_GetPropertyAsIdentity(()``:
+
+.. code-block:: c
+
+    TEE_Result TA_OpenSessionEntryPoint(uint32_t __unused param_types,
+                                        TEE_Param __unused params[4],
+                                        void **tee_session)
+    {
+        TEE_Identity identity = { };
+        TEE_Result res = TEE_SUCCESS;
+
+        res = TEE_GetPropertyAsIdentity(TEE_PROPSET_CURRENT_CLIENT,
+                                        "gpd.client.identity", &identity);
+        if (res)
+            return res;
+
+        switch (identity.login) {
+        case TEE_LOGIN_PUBLIC:
+            return login_public(&identity.uuid, tee_session);
+        case TEE_LOGIN_USER:
+            return login_user(&identity.uuid, tee_session);
+        case TEE_LOGIN_GROUP:
+            return login_group(&identity.uuid, tee_session);
+        case TEE_LOGIN_REE_KERNEL:
+            return login_kernel(&identity.uuid, tee_session);
+        case TEE_LOGIN_TRUSTED_APP:
+            return login_ta(&identity.uuid, tee_session);
+        default:
+            return TEE_ERROR_ACCESS_DENIED;
+        }
+    }
+
+The value of the UUID found in ``identity.uuid`` depends on the login method:
+
+    - When the client is a TA, ``identity.login`` is ``TEE_LOGIN_TRUSTED_APP``
+      and ``identity.uuid`` is the client TA UUID;
+
+    - When the non-secure client uses ``TEE_LOGIN_PUBLIC`` or
+      ``TEE_LOGIN_REE_KERNEL`` method, the UUID is not used. By convention,
+      Linux kernel and U-Boot both set nil UUID (all zeroes).
+
+    - When the non-secure client uses ``TEE_LOGIN_USER`` or ``TEE_LOGIN_GROUP``
+      method, the UUID is generated from the UUIDv5 namespace derivation of a
+      user ID tag (``uid=%x\0``) or a group ID tag (``gid=%x\0``) in
```

Comment:
error: the terminal `\0` is excluded from the tag. I'll also add double-quote `"` to highlight it's a string based ID:
```patch
-      user ID tag (``uid=%x\0``) or a group ID tag (``gid=%x\0``) in
+      user ID tag (``"uid=%x"``) or a group ID tag (``"gid=%x"``) in